src/Workspaces/CoreTestUtilities/Remote/InProcRemostHostClient.cs

```diff
@@ -153,41 +171,122 @@ public object GetService(Type serviceType)
             }
         }
 
+        private sealed class InProcServiceBroker : IServiceBroker
+        {
+            private readonly InProcRemoteServices _services;
+
+            public InProcServiceBroker(InProcRemoteServices services)
+            {
+                _services = services;
+            }
+
+            public event EventHandler<BrokeredServicesChangedEventArgs>? AvailabilityChanged { add { } remove { } }
+
+            public ValueTask<IDuplexPipe?> GetPipeAsync(ServiceMoniker serviceMoniker, ServiceActivationOptions options, CancellationToken cancellationToken)
+                => throw ExceptionUtilities.Unreachable;
+
+            public ValueTask<T?> GetProxyAsync<T>(ServiceRpcDescriptor descriptor, ServiceActivationOptions options, CancellationToken cancellationToken) where T : class
+            {
+                var pipePair = FullDuplexStream.CreatePipePair();
+
+                var clientConnection = descriptor.ConstructRpcConnection(pipePair.Item2);
+
+                Contract.ThrowIfFalse(options.ClientRpcTarget is null == descriptor.ClientInterface is null);
+
+                if (descriptor.ClientInterface != null)
+                {
+                    Contract.ThrowIfNull(options.ClientRpcTarget);
+                    clientConnection.AddLocalRpcTarget(options.ClientRpcTarget);
+                }
+
+                _ = _services.CreateBrokeredService(descriptor, pipePair.Item1, options);
```

Comment:
doc

Reply:
Anything particular you want to have documented here?